rules/aws_s3_bucket_name_length.go

```diff
@@ -0,0 +1,84 @@
+package rules
+
+import (
+	"fmt"
+
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+	"github.com/terraform-linters/tflint-ruleset-aws/project"
+)
+
+// AwsS3BucketNameLengthRule checks that an S3 bucket name contains a valid number of characteres
+type AwsS3BucketNameLengthRule struct {
+	tflint.DefaultRule
+
+	resourceType  string
+	attributeName string
+}
+
+// NewAwsS3BucketNameLengthRule returns new rule with default attributes
+func NewAwsS3BucketNameLengthRule() *AwsS3BucketNameLengthRule {
+	return &AwsS3BucketNameLengthRule{
+		resourceType:  "aws_s3_bucket",
+		attributeName: "bucket",
+	}
+}
+
+// Name returns the rule name
+func (r *AwsS3BucketNameLengthRule) Name() string {
+	return "aws_s3_bucket_name_length"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *AwsS3BucketNameLengthRule) Enabled() bool {
+	return true
+}
+
+// Severity returns the rule severity
+func (r *AwsS3BucketNameLengthRule) Severity() tflint.Severity {
+	return tflint.ERROR
+}
+
+// Link returns the rule reference link
+func (r *AwsS3BucketNameLengthRule) Link() string {
+	return project.ReferenceLink(r.Name())
+}
+
+// Check if s3 bucket name is within the characteres length accepted range
+func (r *AwsS3BucketNameLengthRule) Check(runner tflint.Runner) error {
+
+	resources, err := runner.GetResourceContent(r.resourceType, &hclext.BodySchema{
+		Attributes: []hclext.AttributeSchema{
+			{Name: r.attributeName},
+		},
+	}, nil)
+	if err != nil {
+		return err
+	}
+
+  bucketNameMinLength := 3
+  bucketNameMaxLength := 63
```

Comment:
Can you avoid hardcoding this and find somewhere to reference it? AWS has a lot of schemas for their API that should specify this somewhere...

Reply:
Hmm, seems not:

https://github.com/aws/aws-sdk-go/blob/2ac2bb7f4efd6406c91b30e668024928ba8a63bc/models/apis/s3/2006-03-01/api-2.json#L1350

Reply:
Yeah, it seems rather bad to hardcode, but I also haven't found somewhere to reference it from.